src/messaging/ExchangeMgr.cpp

```diff
@@ -263,7 +270,11 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
     // an ack to the peer.
     else if (!payloadHeader.NeedsAck())
     {
-        ExitNow(err = CHIP_ERROR_UNSOLICITED_MSG_NO_ORIGINATOR);
+        // TODO(#4170): propagate event into device controller, it won't be necessary after fully migrated to messaging layer
```

Comment:
If we have a reserved exchange ID, can we move this check to the beginning of this function?